File: Python/Level 2/가장 큰 수 (정렬).py
def solution(numbers):
    answer = [str(i) for i in numbers]
    
    # 4자리수가 없을 경우 이전 숫자와 비교하기 위해서 x의 길이만큼 나눠줌
    answer.sort(key = lambda x : (x[0],x[1%len(x)],x[2%len(x)],x[3%len(x)]), reverse=True)

    # [0,0,0]의 경우 0이 나와야 하고 답은 str 형식이어야 하므로 str->int->str 형 변환
    return str(int(''.join(answer)))

File: Javascript/Level 2/가장 큰 수(정렬).js
function solution(numbers) {
    numbers.sort((a,b)=>{
        return (String(b)+String(a))-(String(a)+String(b));         // 문자열 b+a 와 a+b 를 비교.
    })
    
    if(numbers[0]===0)                                              // 가장 큰 값이어야 하는 맨 앞 수가 0일 경우 답은 0이다. (000 대신 0을 출력해야 하므로)
        {
            return '0';
        }
    
    return numbers.join('');                                        // 배열 값 합치기
}

// https://programmers.co.kr/learn/courses/30/lessons/42746
// 가장 큰 조합의 수를 찾는 문제